### DISCUSSION TITLE
Problem with case insensitive search

### DISCUSSION THREAD

**Discussion by Maggus85** [alirezanet/Gridify#154] [Q&A] [answered]
Hi 
I'm using Gridify with EF core 8 and cosmos DB.

I created mapper

```
    public EmployeeFilterMapper()
    {
        AddMap("fullName", x => x.LastName + " " + x.FirstName);
        Configuration.CaseSensitive = false;
    }
    
```
i also tried second verion

```
      public EmployeeFilterMapper()
      {
          AddMap("fullName", x => x.LastName + " " + x.FirstName, x => x.ToLower());
          Configuration.CaseSensitive = false;
      } 

```

But it did not change anything,

I wan't to be able to search by a full name ignoring case. When as a filter I send **fullName=*KOW** generted query looks like
    

```
SELECT COUNT(1) AS c
      FROM root c
       WHERE (((c["CosmosEntityName"] = "employees") AND NOT(c["IsDeleted"])) AND CONTAINS(((c["LastName"] || " ") || c["FirstName"]), "KOW"))
```

when as a filter I send **fullName=*KOW/i** generted query looks like, which is ok 
    
  ```
      SELECT COUNT(1) AS c
      FROM root c
      WHERE (((c["CosmosEntityName"] = "employees") AND NOT(c["IsDeleted"])) AND CONTAINS(LOWER(((c["LastName"] || " ") || c["FirstName"])), "kow"))
```
      
Is there is a possiblility co configure mapper, to skip adding /i to a filter?

**Comment by Maggus85** [comment, marked as answer]
I was able to solve it correct mapper should looks like

    public EmployeeFilterMapper()
    {
        AddMap("fullName", x => (x.LastName + " " + x.FirstName).ToLower(), x=> x.ToLower());
    }

**Reply by alirezanet** [reply]
Also, we've added a new feature ([CaseInsensitiveFiltering ](https://alirezanet.github.io/Gridify/guide/gridifyMapper#caseinsensitivefiltering )configuration) starting from version 2.15.0.
